NYPLAudiobookToolkit/Network/AudiobookNetworkService.swift

```diff
@@ -101,6 +102,14 @@ public final class DefaultAudiobookNetworkService: AudiobookNetworkService {
         }
         self.cursor?.currentElement.downloadTask.fetch()
     }
+  
+    public func cancelFetch() {
+        self.cursor = Cursor(data: self.spine)
```

Comment:
this is just my ignorance on this library, but what does `self.cursor` represent and why do we need it? I looked at the code and there's no documentation for that instance variable.

Could we have done this by simply using a local var instead of modifying the state of the AudiobookNetworkService, like 
```
var cursor = Cursor(data: self.spine); 
while cursor != nil 
```
etc ? just wondering, it would be better for thread safety (and I realize this class is not meant to be thread safe from looking at previous code)

Reply:
Yeah that makes sense, nice catch